### ISSUE TITLE
use functional.linear to custom a simple layer

### ISSUE THREAD

**Issue by lith0613** [pytorch/pytorch#7720]
Hi,I try to use functional.linear to custom a simple layer, the weight parameter is combined with two parameters, then the output loss is always nan，I don’t kown how can I sovle this problem or I can’t do that way. my code is as follows:

'''code
class Linear(nn.Module):
    def __init__(self, in_dim,out_dim):
        super(Linear, self).__init__()
        self.weight1 = nn.Parameter(torch.ones(in_dim,out_dim)) 
        self.weight2=nn.Parameter(torch.randn(out_dim,in_dim))
        
    def forward(self, x):
        return F.linear(x, torch.mm(self.weight1,self.weight2))

**Comment by soumith** [comment]
please use https://discuss.pytorch.org
